### PR TITLE
[BEAM-1280] Remove fn_or_label argument from Map and Filter

### PR DESCRIPTION
**Pull request by sb2nov**
Part 1/? for removing the fn_or_label functionality from ptransforms

R: @aaltay PTAL

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -721,15 +720,10 @@ def Map(fn_or_label, *args, **kwargs):  # pylint: disable=invalid-name
...
-  if isinstance(fn_or_label, str):
-    label, fn, args = fn_or_label, args[0], args[1:]
-  else:
-    label, fn = None, fn_or_label
   if not callable(fn):
     raise TypeError(
         'Map can be used only with callable objects. '
-        'Received %r instead for %s argument.'
-        % (fn, 'first' if label is None else 'second'))
+        'Received %r instead' % (fn))
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/core.py]: add a `.` at the end of the sentence.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -742,25 +736,23 @@ def Map(fn_or_label, *args, **kwargs):  # pylint: disable=invalid-name
   # Proxy the type-hint information from the original function to this new
   # wrapped function.
   get_type_hints(wrapper).input_types = get_type_hints(fn).input_types
-  output_hint = get_type_hints(fn).simple_output_type(label)
+  output_hint = get_type_hints(fn).simple_output_type(None)
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/core.py]: why not `simple_output_type()` ? I suppose we will remove the argument from `simple_output_type` as well.

File: sdks/python/apache_beam/transforms/core.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/core.py]: Done

File: sdks/python/apache_beam/transforms/core.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/core.py]: Done. simple_output_type uses the label to create a clean error so we should propagate the default label here.

**Comment by aaltay** [comment]
LGTM. Thank you.

**Pull request closed without merging**